stitch/src/main/java/com/mongodb/stitch/android/auth/custom/CustomAuthProviderInfo.java

```diff
@@ -0,0 +1,97 @@
+package com.mongodb.stitch.android.auth.custom;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mongodb.stitch.android.auth.AuthProviderInfo;
+import com.mongodb.stitch.android.auth.oauth2.google.GoogleAuthProviderInfo;
+
+import java.util.List;
+
+/**
+ * Created by jasonflax on 12/4/17.
+ */
+
+public class CustomAuthProviderInfo extends AuthProviderInfo {
+    public static final String FQ_NAME = "custom-token";
+    private static final String METADATA_FIELDS = "metadata_fields";
+    private static final String CONFIG = "config";
+
+    public static class MetadataField {
```

Comment:
I feel like this belongs in its own class, not an inner static class

Reply:
I was under the impression, initially, that MetadataField could be unique to a provider. If that's not the case, then it can be its own class.